src/Protocol/Models/TextDocumentIdentifier.cs

```diff
@@ -4,44 +4,13 @@
 namespace OmniSharp.Extensions.LanguageServer.Protocol.Models
 {
     [DebuggerDisplay("{" + nameof(DebuggerDisplay) + ",nq}")]
-    public class TextDocumentIdentifier : IEquatable<TextDocumentIdentifier>
+    public record TextDocumentIdentifier(DocumentUri Uri)
     {
-        public TextDocumentIdentifier()
-        {
-        }
+        public TextDocumentIdentifier(): this(null!) {}
 
-        public TextDocumentIdentifier(DocumentUri uri) => Uri = uri;
-
-        /// <summary>
-        /// The text document's URI.
-        /// </summary>
-        public DocumentUri Uri { get; set; } = null!;
-
-        public bool Equals(TextDocumentIdentifier? other)
-        {
-            if (ReferenceEquals(null, other)) return false;
-            if (ReferenceEquals(this, other)) return true;
-            return Uri.Equals(other.Uri);
-        }
-
-        public override bool Equals(object? obj)
```

Comment:
Lots of boilerplate being removed here, pretty nice considering I believe that was a big goal of records.